### PR TITLE
Fix oc process -o template --template <template>

### PR DESCRIPTION
**Pull request by mmilata**
Before:
```
$ oc process something -o template --template {{.Kind}}
error: template format specified but no template given
```
After:
```
$ oc process something -o template --template {{.Kind}}
List
```

@bparees PTAL

### CODE REVIEW

**Comment by mmilata** [comment]
Alternatives:
* remove `--template`
* use `-o go-template={{.something}}` like `oc get` does

**Comment by bparees** [comment]
let's make it go-template so we're aligned.

**Comment by mmilata** [comment]
Pushed new version that supports go-template and jsonpath. [test]

File: pkg/cmd/cli/cmd/process.go [hunk 1]
```diff
@@ -78,10 +78,18 @@ func NewCmdProcess(fullName string, f *clientcmd.Factory, out, errout io.Writer)
...
+	cmd.Flags().StringP("output", "o", "json", "Output format. One of: describe|json|yaml|name|go-template=...|go-template-file=...|jsonpath=...|jsonpath-file=...")
 	cmd.Flags().Bool("raw", false, "If true output the processed template instead of the template's objects. Implied by -o describe")
 	cmd.Flags().String("output-version", "", "Output the formatted object with the given version (default api-version).")
-	cmd.Flags().StringP("template", "t", "", "Template string or path to template file to use when -o=template or -o=templatefile.  The template format is golang templates [http://golang.org/pkg/text/template/#pkg-overview]")
+	cmd.Flags().StringP("template", "t", "", "Template string or path to template file to use when -o=go-template, -o=go-templatefile.  The template format is golang templates [http://golang.org/pkg/text/template/#pkg-overview]")
+
+	// kcmdutil.PrinterForCommand needs these flags, however they are useless
+	// here because oc process returns list of heterogeneous objects that is
+	// not suitable for formatting as a table.
+	cmd.Flags().Bool("no-headers", false, "When using the default output, don't print headers.")
```
Comment (bparees) [inline pkg/cmd/cli/cmd/process.go]: i would have expected this to be true?

File: test/cmd/templates.sh [hunk 2]
```diff
@@ -22,6 +22,17 @@ os::cmd::expect_success 'oc get templates'
 os::cmd::expect_success 'oc get templates ruby-helloworld-sample'
 os::cmd::expect_success 'oc get template ruby-helloworld-sample -o json | oc process -f -'
 os::cmd::expect_success 'oc process ruby-helloworld-sample'
+os::cmd::expect_success_and_text 'oc process ruby-helloworld-sample -o template --template "{{.kind}}"'    "List"
```
Comment (bparees) [inline test/cmd/templates.sh]: -o template still works?

Comment (mmilata) [inline test/cmd/templates.sh]: Yes. Like in `oc get` it is not shown in help but still works. I suppose it is deprecated in favor of `-o go-template`.

File: pkg/cmd/cli/cmd/process.go (hunk 1 above)
Comment (mmilata) [inline pkg/cmd/cli/cmd/process.go]: I think the value does not really matter, let me check.

Comment (bparees) [inline pkg/cmd/cli/cmd/process.go]: ok

**Comment by bparees** [comment]
lgtm [merge]

**Pull request merged**